src/test/ui/consts/min_const_fn/min_const_fn_unsafe_bad.stderr

```diff
@@ -25,14 +25,14 @@ LL | const unsafe fn bad_const_unsafe_deref_raw_ref(x: *mut usize) -> &'static u
    = note: for more information, see https://github.com/rust-lang/rust/issues/51911
    = help: add `#![feature(const_raw_ptr_deref)]` to the crate attributes to enable
 
-error[E0658]: unions in const fn are unstable
+error[E0723]: accessing union fields is unstable
   --> $DIR/min_const_fn_unsafe_bad.rs:14:5
    |
 LL |     Foo { x: () }.y
    |     ^^^^^^^^^^^^^^^
    |
-   = note: for more information, see https://github.com/rust-lang/rust/issues/51909
-   = help: add `#![feature(const_fn_union)]` to the crate attributes to enable
```

Comment:
What's up with the feature gate changing here? I assume this is solely due to the change in `check_unsafety` but this seems like an unfortunate change.

Reply:
You need the const_fn feature gate. Once you enable that, you'll get told about `const_fn_union`

Reply:
Oh; that's ungreat UX but not worth the effort to fix. Do we have a test which checks that once you do have `const_fn` active you also get complaints re. `const_fn_union`?

Reply:
Yes, we have https://github.com/rust-lang/rust/blob/master/src/test/ui/consts/const-eval/feature-gate-const_fn_union.rs

Reply:
Excellent.